src/dex/weth/weth.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  Token,
+  Address,
+  ExchangePrices,
+  AdapterExchangeParam,
+  SimpleExchangeParam,
+  PoolLiquidity,
+  Logger,
+} from '../../types';
+import { SwapSide, Network } from '../../constants';
+import {
+  getDexKeysWithNetwork,
+  isETHAddress,
+  WethMap,
+  isWETH,
+} from '../../utils';
+import { IDex } from '../../dex/idex';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import {
+  WethData,
+  WethFunctions,
+  DexParams,
+  IWethDepositorWithdrawer,
+  DepositWithdrawReturn,
+} from './types';
+import { SimpleExchange } from '../simple-exchange';
+import { Adapters, WethConfig } from './config';
+
+export class Weth
+  extends SimpleExchange
+  implements IDex<WethData, DexParams>, IWethDepositorWithdrawer
+{
+  readonly hasConstantPriceLargeAmounts = true;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(WethConfig);
+
+  public static getAddress(network: number = 1): Address {
+    return WethMap[network];
+  }
+
+  logger: Logger;
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+    protected adapters = Adapters[network] || {},
+    protected unitPrice = BigInt(1e18),
+    protected poolGasCost = WethConfig[dexKey][network].poolGasCost,
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {}
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] || null;
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifiers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (srcToken.address.toLowerCase() === destToken.address.toLowerCase()) {
+      return [];
+    }
+
+    const tokenAddress = [
+      srcToken.address.toLowerCase(),
+      destToken.address.toLowerCase(),
+    ]
+      .sort((a, b) => (a > b ? 1 : -1))
+      .join('_');
+
+    const poolIdentifier = `${this.dexKey}_${tokenAddress}`;
+    return [poolIdentifier];
```

Comment:
I don't this is correct. We should check if it is WETH <> ETH swap. If not return empty list. If it is then the poolIdentifier should be Network_WETHAddress

Reply:
Fixed. Check please